packages/config/src/main.ts

```diff
@@ -42,7 +42,9 @@ export const resolveConfig = async function (opts) {
   const api = getApiClient({ token, offline, host, scheme, pathPrefix, testOpts })
 
   const parsedCachedConfig = await getCachedConfig({ cachedConfig, cachedConfigPath, token, api })
-  if (parsedCachedConfig !== undefined) {
+  // if we receive a defaultConfig (something the CLI can set for the runBuild with defaults)
+  // we need to invalidate the cached config
```

Comment:
The change makes sense to me, but I find the comment confusing. We're not invalidating the cached config, we're just giving precedence to the default config.

Here's a suggestion for how that might be made more clear:

```suggestion
  // If there is a cached config, use it. The exception is when a default config,
  // which consumers like the CLI can set, is present. In those cases, let the
  // flow continue so that the default config is parsed and used.
```

Reply:
Oh that's a great suggestion!